model/atmosphere/dycore/tests/dycore_stencil_tests/test_compute_perturbed_quantities_and_interpolation.py

```diff
@@ -0,0 +1,443 @@
+            ),
+            (perturbed_theta_v_at_cells_on_half_levels, theta_v_at_cells_on_half_levels),
+        )
+
+        if igradp_method == 3:
+            d2dz2_of_temporal_extrapolation_of_perturbed_exner_on_model_levels = np.where(
+                (start_cell_lateral_boundary_level_3 <= horz_idx)
+                & (horz_idx < end_cell_halo)
+                & (nflat_gradp <= vert_idx[:n_lev]),
+                compute_approx_of_2nd_vertical_derivative_of_exner_numpy(
+                    z_theta_v_pr_ic=perturbed_theta_v_at_cells_on_half_levels,
+                    d2dexdz2_fac1_mc=d2dexdz2_fac1_mc,
+                    d2dexdz2_fac2_mc=d2dexdz2_fac2_mc,
+                    z_rth_pr_2=perturbed_theta_v_at_cells_on_model_levels[:, :n_lev],
+                ),
+                d2dz2_of_temporal_extrapolation_of_perturbed_exner_on_model_levels,
+            )
+
+        (
+            perturbed_rho_at_cells_on_model_levels,
+            perturbed_theta_v_at_cells_on_model_levels[:, :n_lev],
+        ) = np.where(
+            (start_cell_halo_level_2 <= horz_idx) & (horz_idx < end_cell_halo_level_2),
+            compute_perturbation_of_rho_and_theta_numpy(
+                rho=current_rho,
+                rho_ref_mc=reference_rho_at_cells_on_model_levels,
+                theta_v=current_theta_v,
+                theta_ref_mc=reference_theta_at_cells_on_model_levels,
+            ),
+            (
+                perturbed_rho_at_cells_on_model_levels,
+                perturbed_theta_v_at_cells_on_model_levels[:, :n_lev],
+            ),
+        )
+
+        return dict(
+            temporal_extrapolation_of_perturbed_exner=temporal_extrapolation_of_perturbed_exner,
+            perturbed_exner_at_cells_on_model_levels=perturbed_exner_at_cells_on_model_levels,
+            exner_at_cells_on_half_levels=exner_at_cells_on_half_levels,
+            ddz_of_temporal_extrapolation_of_perturbed_exner_on_model_levels=ddz_of_temporal_extrapolation_of_perturbed_exner_on_model_levels,
+            perturbed_rho_at_cells_on_model_levels=perturbed_rho_at_cells_on_model_levels,
+            perturbed_theta_v_at_cells_on_model_levels=perturbed_theta_v_at_cells_on_model_levels,
+            rho_at_cells_on_half_levels=rho_at_cells_on_half_levels,
+            perturbed_theta_v_at_cells_on_half_levels=perturbed_theta_v_at_cells_on_half_levels,
+            theta_v_at_cells_on_half_levels=theta_v_at_cells_on_half_levels,
+            pressure_buoyancy_acceleration_at_cells_on_half_levels=pressure_buoyancy_acceleration_at_cells_on_half_levels,
+            d2dz2_of_temporal_extrapolation_of_perturbed_exner_on_model_levels=d2dz2_of_temporal_extrapolation_of_perturbed_exner_on_model_levels,
+        )
+
+    @pytest.fixture
+    def input_data(self, grid: base.BaseGrid) -> dict[str, gtx.Field | state_utils.ScalarType]:
+        reference_rho_at_cells_on_model_levels = data_alloc.random_field(grid, CellDim, KDim)
```

Comment:
```suggestion
        reference_rho_at_cells_on_model_levels = data_alloc.random_field(grid, dims.CellDim, dims.KDim)
```
for all of the others as well, import dimension file as `dims`, and remove CellDim, KDim direct import